contracts/external/dao-migrator/src/contract.rs

```diff
@@ -88,18 +88,23 @@ fn execute_migration_v1_v2(
     }
 
     // List of code ids pairs we got and the migration msg of each one of them.
-    let proposal_pairs: Vec<CodeIdPair> = migration_params
+    let proposal_pairs: Vec<(String, CodeIdPair)> = migration_params
         .proposal_params
```

Comment:
just a little thing, but could we add a check that this does not have duplicate `addr` values?

Reply:
```rust 
let mut uniq = HashSet::new();
if migration_params.proposal_params.iter().all(|(addr, _)| uniq.insert(addr)) {
    return Err(ContractError::DuplicateProposalParams);
}
```

Probably the nicest way I found to check for duplicates, not sure if HashSet is supported on the blockchain (it should in theory)

Reply:
looks great!